biosys/apps/main/api/uploaders.py

```diff
@@ -63,17 +63,40 @@ class FileReader(object):
     ]
     SUPPORTED_TYPES = CSV_TYPES + XLSX_TYPES
 
-    def __init__(self, file_):
-        if file_.content_type not in self.SUPPORTED_TYPES:
-            msg = "Wrong file type {}. Should be one of: {}".format(file_.content_type, self.SUPPORTED_TYPES)
-            raise Exception(msg)
+    CSV_FORMAT = 'csv'
+    XLSX_FORMAT = 'xlsx'
+    NOT_SUPPORTED_FORMAT = 'not supported'
+
+    @staticmethod
+    def get_uploaded_file_format(uploaded_file):
+        """
+        Return 'csv', 'xlsx' or 'not supported'
+        :param uploaded_file: a Django uploaded file
+        :return:
+        """
+        if not hasattr(uploaded_file, 'name') or not hasattr(uploaded_file, 'content_type'):
+            raise Exception("The given file is not a django managed uploaded file: {}".format(uploaded_file))
+        # check extension first
+        # Note: on Windows the content-type for a csv file can be sent with an excel mime type.
+        file_name = uploaded_file.name
+        content_type = uploaded_file.content_type
+        extension = path.splitext(file_name)[1].lower()
+        result = FileReader.NOT_SUPPORTED_FORMAT
```

Comment:
for clarity i'd put this line in an else statement below

Reply:
I did it like that first.